material/elements/vaadin-date-picker.html

```diff
@@ -0,0 +1,218 @@
+      [part~="date"] {
+        position: relative;
+        height: 32px;
+        font-weight: normal;
+
+        display: flex;
+        align-items: center;
+        justify-content: center;
+      }
+
+      [part~="date"][focused]:not([disabled])::after {
+        position: absolute;
+        top: 0;
+        bottom: 0;
+        left: 0;
+        right: 0;
+        margin: auto;
+        z-index: -1;
+        content: '';
+        height: 32px;
+        width: 32px;
+        border-radius: 100%;
+        opacity: 0.5;
+        background: var(--material-primary-color);
+      }
+
+      :host([focused]) [part~="date"][focused]:not([disabled])::after {
+        opacity: 1;
+      }
+
+      [part~="date"][today] {
+        color: var(--material-primary-text-color);
+      }
+
+      [part~="date"][selected] {
+        font-weight: bold;
+      }
+
+      :host([focused]) [part~="date"][focused]:not([disabled]) {
+        background: transparent;
+      }
+
+      [part~="date"][focused]:not([disabled]) {
+        color: var(--material-primary-contrast-color);
+        background: transparent;
+        box-sizing: border-box;
+      }
+
+      [part~="date"][selected] {
+        font-weight: 600;
+      }
+
+      [part~="weekdays"] {
+        color: rgba(0, 0, 0, 0.5);
+        font-size: small;
+      }
+
+      [part~="weekday"] {
+        color: var(--material-secondary-text-color);
+      }
+
+      [part~="week-number"] {
+        height: 32px;
+        font-size: var(--material-font-size-s);
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        color: var(--material-disabled-text-color);
+      }
+
+      [part~="month-header"] {
+        font-size: var(--material-font-size-l);
+        font-weight: var(--material-font-weight-strong);
+      }
+    </style>
+  </template>
+</dom-module>
+<dom-module id="material-date-picker-overlay" theme-for="vaadin-date-picker-overlay">
+  <template>
+    <style include="vaadin-date-picker-overlay-default-theme">
+       :host() {
+        border-radius: 2px;
+        border: 1px solid rgba(0, 0, 0, 0.3);
+        background: var(--material-background-color);
+      }
+
+      :host(:not([fullscreen])) {
+        box-shadow: 0 2px 4px 2px rgba(0, 0, 0, 0.3);
+
+        /* tweak the margin-top set by iron-dropdown. */
+        transform: translateY(-2px);
+      }
+
+      [part~="years"] {
+        background: var(--material-secondary-background-color);
+        z-index: 1;
+      }
+
+      [part~="years"]::before {
+        border-left-color: var(--material-background-color);
+      }
+
+      [part~="years"]::after {
+        content: '';
+        position: absolute;
+        top: 0;
+        left: 0;
+        width: 4px;
+        height: 100%;
+        background: linear-gradient(.25turn, rgba(0, 0, 0, 0.1), rgba(0, 0, 0, 0));
+      }
+
+      [part~="months"] {
+        --vaadin-infinite-scroller-item-height: 300px;
```

Comment:
This should be based on `--material-base-grid`. Perhaps we need to fix the height of the `month-header` and `weekdays` elements using that as well?